redwood-treehouse-host/src/iosMain/kotlin/app/cash/redwood/treehouse/TreehouseUIView.kt

```diff
@@ -50,26 +54,41 @@ public class TreehouseUIView private constructor(
   public constructor(widgetSystem: WidgetSystem<UIView>) : this(widgetSystem, RootUiView())
 
   init {
-    (view as RootUiView).treehouseView = this
+    view.treehouseView = this
   }
 
   private fun superviewChanged() {
     readyForContentChangeListener?.onReadyForContentChanged(this)
   }
 
-  private class RootUiView : UIView(cValue { CGRectZero }) {
+  /**
+   * The root view is just a vertical stack.
+   *
+   * In practice we expect this to contain either zero child subviews (especially when
+   * newly-initialized) or one child subview, which will usually be a layout container.
+   *
+   * This could just as easily be a horizontal stack. A box would be even better, but there's no
+   * such built-in component and implementing it manually is difficult if we want to react to
+   * content resizes.
+   */
+  private class RootUiView : UIStackView(cValue { CGRectZero }) {
```

Comment:
Using a `UIStackView` when we expect it to always have at most one subview doesn't feel ideal, but if it helps with resizing the child view, it works.

Reply:
Not sure if Treehouse is using autolayout or frame-based layout, we have definitely run into issue before with `UIStackView` and frame-based layout.

Reply:
I believe we're mostly using frame-based layout within this repo--In the widget bindings defined `cash-ios`, we use paralayout, which is frame-based as well.